Lib/functools.py

```diff
@@ -447,54 +443,93 @@ class partialmethod:
-        _method.__partialmethod__ = self
-        return _method
+    def __init__(self, func, /, *args, **keywords):
+        if isinstance(func, partialmethod):
+            # Subclass optimization
+            temp = partial(lambda: None, *func.args, **func.keywords)
+            temp = partial(temp, *args, **keywords)
+            func = func.func
+            args = temp.args
+            keywords = temp.keywords
+
+        self.func = func
+        self.args = args
+        self.keywords = keywords
+
+        if isinstance(func, _STD_METHOD_TYPES):
+            self.method = None
+        elif getattr(func, '__get__', None) is None:
+            if not callable(func):
+                raise TypeError(f'the first argument {func!r} must be a callable '
+                                'or a descriptor')
+            self.method = None
+        else:
+            # Unknown descriptor
+            self.method = _UNKNOWN_DESCRIPTOR
+
+    def _set_func_attrs(self, func):
+        func.__partialmethod__ = self
+        if self.__isabstractmethod__:
+            func = abstractmethod(func)
+        return func
+
+    def _make_method(self):
+        args = self.args
+        func = self.func
+
+        if isinstance(func, staticmethod):
+            deco = staticmethod
+            method = partial(func.__wrapped__, *args, **self.keywords)
+        elif isinstance(func, classmethod):
+            deco = classmethod
+            ph_args = _ONE_PLACEHOLDER_TUPLE if args else ()
+            method = partial(func.__wrapped__, *ph_args, *args, **self.keywords)
+        else:
+            # instance method. 2 cases:
+            #   a) FunctionType | partial
+            #   b) callable object without __get__
+            deco = None
+            ph_args = _ONE_PLACEHOLDER_TUPLE if args else ()
+            method = partial(func, *ph_args, *args, **self.keywords)
+
+        method.__partialmethod__ = self
+        if self.__isabstractmethod__:
+            method = abstractmethod(method)
+        if deco is not None:
+            method = deco(method)
+        return method
 
     def __get__(self, obj, cls=None):
-        get = getattr(self.func, "__get__", None)
-        result = None
-        if get is not None:
-            new_func = get(obj, cls)
-            if new_func is not self.func:
-                # Assume __get__ returning something new indicates the
-                # creation of an appropriate callable
-                result = partial(new_func, *self.args, **self.keywords)
-                try:
-                    result.__self__ = new_func.__self__
-                except AttributeError:
-                    pass
-        if result is None:
-            # If the underlying descriptor didn't do anything, treat this
-            # like an instance method
-            result = self._make_unbound_method().__get__(obj, cls)
-        return result
+        method = self.method
+        if method is _UNKNOWN_DESCRIPTOR:
```

Comment:
Instead of this additional _UNKNOWN_DESCRIPTOR, can't you just check `not hasattr(self, 'method')`